core/src/test/java/io/questdb/test/griffin/AsOfJoinFuzzTest.java

```diff
@@ -84,13 +63,41 @@ public void testFuzzSomeDuplicates() throws Exception {
         testFuzz(10);
     }
 
-    private void assertResultSetsMatch() throws Exception {
+    private void assertResultSetsMatch0(Rnd rnd) throws Exception {
+        Object[][] allParameterPermutations = TestUtils.cartesianProduct(new Object[][]{
+                JoinType.values(),
+                {true, false}, // exercise interval intrinsics
+                LimitType.values(),
+                {true, false}, // exercise filters
+                ProjectionType.values(),
+        });
+        try {
+            for (int i = 0; i < allParameterPermutations.length; i++) {
+                Object[] params = allParameterPermutations[i];
+                JoinType joinType = (JoinType) params[0];
+                boolean exerciseIntervals = (boolean) params[1];
+                LimitType limitType = (LimitType) params[2];
+                boolean exerciseFilters = (boolean) params[3];
+                ProjectionType projectionType = (ProjectionType) params[4];
+                assertResultSetsMatch0(joinType, exerciseIntervals, limitType, exerciseFilters, projectionType, rnd);
+            }
+        } catch (AssertionError e) {
+            throw new AssertionError("Failed with parameters: " +
```

Comment:
Why do we assume that the zero array element always fails? Should be `i`-th one, no?

Reply:
doh